packages/marketplace/contracts/TransferManager.sol

```diff
@@ -199,18 +211,107 @@ abstract contract TransferManager is Initializable, ITransferManager {
+                uint256 idLength = bundle.bundledERC1155[i].ids.length;
+                require(idLength == bundle.bundledERC1155[i].supplies.length, "ERC1155 array error");
+                for (uint256 j; j < idLength; j++) {
+                    uint256 tokenId = bundle.bundledERC1155[i].ids[j];
+                    IRoyaltiesProvider.Part[] memory royalties = royaltiesRegistry.getRoyalties(token, tokenId);
+                    for (uint256 k; k < royalties.length; k++) {
+                        royalties[i].basisPoints = royalties[i].basisPoints * bundle.bundledERC1155[i].supplies[j]; // cumulative basis point
+                    }
+                    remainder = _applyRoyalties(
+                        remainder,
+                        paymentSide,
+                        nftSide.asset.priceDistribution.erc1155Prices[i][j],
+                        royalties,
+                        nftSide.recipient
+                    );
+                }
+            }
+
+            uint256 quadSize = bundle.quads.xs.length;
+            if (quadSize > 0) {
+                for (uint256 i = 0; i < quadSize; i++) {
+                    uint256 size = bundle.quads.sizes[i];
+                    uint256 x = bundle.quads.xs[i];
+                    uint256 y = bundle.quads.ys[i];
+                    IRoyaltiesProvider.Part[] memory quadRoyalties = new IRoyaltiesProvider.Part[](1);
+                    for (uint256 j = 0; j < size * size; j++) {
+                        uint256 tokenId = idInPath(j, size, x, y);
+                        IRoyaltiesProvider.Part[] memory royalties = royaltiesRegistry.getRoyalties(
+                            address(landContract),
+                            tokenId
+                        );
+
+                        for (uint256 k; k < royalties.length; k++) {
+                            quadRoyalties[0].basisPoints += royalties[k].basisPoints; // cumulative basis point of all land
+                            quadRoyalties[0].account = royalties[k].account;
+                        }
+                    }
+
+                    remainder = _applyRoyalties(
+                        remainder,
+                        paymentSide,
+                        nftSide.asset.priceDistribution.quadPrices[i],
+                        quadRoyalties,
+                        nftSide.recipient
+                    );
+                }
+            }
+        } else {
+            (address token, uint256 tokenId) = LibAsset.decodeToken(nftSide.asset.assetType);
+            IRoyaltiesProvider.Part[] memory royalties = royaltiesRegistry.getRoyalties(token, tokenId);
+            remainder = _applyRoyalties(remainder, paymentSide, remainder, royalties, nftSide.recipient);
+        }
+        return remainder;
+    }
+
+    function _applyRoyalties(
+        uint256 remainder,
+        DealSide memory paymentSide,
+        uint256 assetPrice,
+        IRoyaltiesProvider.Part[] memory royalties,
+        address recipient
```

Comment:
What is the point of adding another parameter `address recipient` when you always pass `nftSide.recipient`?

Reply:
when calling _applyRoyalties() the address of recipient from nftSide is passed instead of nftSide

Reply:
I think I see now, you made a new `_applyRoyalties` function that you call inside _transferRoyalties and I got confused about it. All good here.